contrib/graph-gophers/graphql-go/graphql.go

```diff
@@ -0,0 +1,72 @@
+package graphql
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/graph-gophers/graphql-go/errors"
+	"github.com/graph-gophers/graphql-go/introspection"
+	"github.com/graph-gophers/graphql-go/trace"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/tracer"
+)
+
+const (
+	graphqlFieldTag = "graphql.field"
+	graphqlQueryTag = "graphql.query"
+	graphqlTypeTag  = "graphql.type"
+)
+
+// A Tracer implements the graphql-go/trace.Tracer interface by sending traces
+// to the Datadog tracer.
+type Tracer struct {
+	cfg *config
+}
+
+// TraceQuery traces a GraphQL query.
+func (t *Tracer) TraceQuery(ctx context.Context, queryString string, operationName string, variables map[string]interface{}, varTypes map[string]*introspection.Type) (context.Context, trace.TraceQueryFinishFunc) {
+	span, ctx := tracer.StartSpanFromContext(ctx, "graphql.request",
+		tracer.ServiceName(t.cfg.serviceName),
+		tracer.Tag(graphqlQueryTag, queryString),
+	)
+
+	return ctx, func(errs []*errors.QueryError) {
+		var err error
```

Comment:
We could simplify this, one idea is:
```go
switch n := len(errs); n {
    case 0:
        // err = nil
    case 1:
        err = errs[0]
    default:
        err = fmt.Errorf("%s (and %d more errors)", errs[0], n-1)
}
```
The `fmt` package formatter `%s` will correctly call the `error` interface's `Error` method to obtain the string.